go-controller/pkg/ovn/controller/admin_network_policy/baseline_admin_network_policy.go

```diff
@@ -0,0 +1,259 @@
+	c.Lock()
+	defer c.Unlock()
+	startTime := time.Now()
+	_, banpName, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+	klog.V(4).Infof("Processing sync for Baseline Admin Network Policy %s", banpName)
+
+	defer func() {
+		klog.V(4).Infof("Finished syncing Baseline Admin Network Policy %s : %v", banpName, time.Since(startTime))
+	}()
+
+	banp, err := c.banpLister.Get(banpName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	if banp == nil {
+		// it was deleted; let's clear up all the related resources to that
+		err = c.clearBaselineAdminNetworkPolicy(banpName)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	// at this stage the BANP exists in the cluster
+	err = c.ensureBaselineAdminNetworkPolicy(banp)
+	if err != nil {
+		// we can ignore the error if status update doesn't succeed; best effort
+		_ = c.updateBANPStatusToNotReady(banp, c.zone, err.Error())
+		return err
+	}
+	// we can ignore the error if status update doesn't succeed; best effort
+	_ = c.updateBANPStatusToReady(banp, c.zone)
+	return nil
+}
+
+// clearBaselineAdminNetworkPolicy will handle the logic for deleting all db objects related
+// to the provided banp which got deleted.
+// uses externalIDs to figure out ownership
+func (c *Controller) clearBaselineAdminNetworkPolicy(banpName string) error {
+	banp := c.banpCache
+	if banp.name != banpName {
+		// there is no existing BANP configured with this name, nothing to clean
+		klog.Infof("BANP %s/%d not found in cache, nothing to clear", banpName, BANPFlowPriority)
+		return nil
+	}
+
+	// clear NBDB objects for the given BANP (PG, ACLs on that PG, AddrSets used by the ACLs)
+	// remove PG for Subject (ACLs will get cleaned up automatically)
+	portGroupName, readableGroupName := getAdminNetworkPolicyPGName(banp.name, true)
+	// no need to batch this with address-set deletes since this itself will contain a bunch of ACLs that need to be deleted which is heavy enough.
+	err := libovsdbops.DeletePortGroups(c.nbClient, portGroupName)
+	if err != nil {
+		return fmt.Errorf("unable to delete PG %s for BANP %s: %w", readableGroupName, banp.name, err)
+	}
+	// remove address-sets that were created for the peers of each rule fpr the whole ANP
+	// do this after ACLs are gone so that there is no lingering references
+	err = c.clearASForPeers(banp.name, libovsdbops.AddressSetBaselineAdminNetworkPolicy)
+	if err != nil {
+		return fmt.Errorf("failed to delete address-sets for BANP %s: %w", banp.name, err)
+	}
+	// we can delete the object from the cache now (set the cache back to empty value).
+	c.banpCache = &adminNetworkPolicyState{}
+
+	return nil
+}
+
+// ensureBaselineAdminNetworkPolicy will handle the main reconcile logic for any given banp's
+// add/update that might be triggered either due to BANP changes or the corresponding
+// matching pod or namespace changes.
+func (c *Controller) ensureBaselineAdminNetworkPolicy(banp *anpapi.BaselineAdminNetworkPolicy) error {
```

Comment:
As a general comment here, I think that there are chances to share a lot more code between ANP and BANP processing. I guess that's something we can do in the FUP.